Packages/Tracking/Physical Hands/Runtime/Scripts/PhysicalHandsButtonBase.cs

```diff
@@ -0,0 +1,344 @@
+        [Space(3)]
+        public UnityEvent OnButtonPressed;
+        public UnityEvent OnButtonUnPressed;
+        [Header("Button Contact")]
+        [Space(3)]
+        public UnityEvent<ContactHand> OnHandContact;
+        public UnityEvent<ContactHand> OnHandContactExit;
+        [Header("Button Hover")]
+        [Space(3)]
+        public UnityEvent<ContactHand> OnHandHover;
+        public UnityEvent<ContactHand> OnHandHoverExit;
+
+        private void OnEnable()
+        {
+            Initialize();
+        }
+
+        /// <summary>
+        /// Initialize the button.
+        /// </summary>
+        private void Initialize()
+        {
+            if (_pressableObject == null)
+            {
+                Debug.LogError("Pressable object not assigned. Please assign one to use the button.");
+                enabled = false;
+                return;
+            }
+
+            _rigidbody = GetComponent<Rigidbody>();
+            _rigidbody.isKinematic = true;
+            _rigidbody.constraints = RigidbodyConstraints.FreezeAll;
+
+            // Set default travel distance if one is not set
+            if (_buttonTravelDistance == 0)
+            {
+                // leave 1mm so the button does not clip into the base obect (assuming same thickness)
+                _buttonTravelDistance = Mathf.Abs(_pressableObject.transform.localPosition.y) - 0.001f; 
+            }
+
+            SetUpPressableObject();
+            SetUpSpringJoint();
+            _initialButtonPosition = _pressableObject.transform.localPosition;
+        }
+
+        /// <summary>
+        /// Set up the pressable object.
+        /// </summary>
+        private void SetUpPressableObject()
+        {
+            // Ensure the pressable object has a Rigidbody component
+            if (!_pressableObject.TryGetComponent<Rigidbody>(out _pressableObjectRB))
+            {
+                _pressableObjectRB = _pressableObject.AddComponent<Rigidbody>();
+                _pressableObjectRB.useGravity = false;
+            }
+
+
+
+            // Ensure the pressable object has an IgnorePhysicalHands component
+            if (!_pressableObject.TryGetComponent<IgnorePhysicalHands>(out IgnorePhysicalHands ignorePhysHands))
+            {
+                ignorePhysHands = _pressableObject.AddComponent<IgnorePhysicalHands>();
+                ignorePhysHands.DisableAllGrabbing = true;
+                ignorePhysHands.DisableAllHandCollisions = false;
+                ignorePhysHands.DisableCollisionOnChildObjects = false;
+            }
+
+            // Ensure the pressable object has a PhysicalHandsButtonHelper component
+            if (!_pressableObject.TryGetComponent<PhysicalHandsButtonHelper>(out _buttonHelper))
+            {
+                _buttonHelper = _pressableObject.AddComponent<PhysicalHandsButtonHelper>();
+            }
+
+            // Subscribe to events from the button helper
+            _buttonHelper._onHandContact += OnHandContactPO;
```

Comment:
consider also unsubbing to the events first, to avoid double subbing when things break